test/e2e/apigatewayv2/service_bootstrap.py

```diff
@@ -0,0 +1,95 @@
+# Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You may
+# not use this file except in compliance with the License. A copy of the
+# License is located at
+#
+#	 http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+"""Bootstraps the resources required to run the APIGatewayV2 integration tests.
+"""
+import logging
+import os
+import time
+from zipfile import ZipFile
+
+import boto3
+
+from apigatewayv2.bootstrap_resources import (
+    TestBootstrapResources,
+    AUTHORIZER_IAM_ROLE_NAME,
+    AUTHORIZER_ASSUME_ROLE_POLICY,
+    AUTHORIZER_POLICY_ARN,
+    AUTHORIZER_FUNCTION_NAME
+)
+from common.aws import get_aws_region
+
+
+def service_bootstrap() -> dict:
+    logging.getLogger().setLevel(logging.INFO)
+    authorizer_role_arn = create_authorizer_role()
+    time.sleep(15)
+    authorizer_function_arn = create_lambda_authorizer(authorizer_role_arn)
+
+    return TestBootstrapResources(
+        authorizer_role_arn,
+        authorizer_function_arn
+    ).__dict__
+
+
+def create_authorizer_role() -> str:
+    region = get_aws_region()
+    iam_client = boto3.client("iam", region_name=region)
+
+    logging.debug(f"Creating authorizer iam role {AUTHORIZER_IAM_ROLE_NAME}")
+
+    try:
+        iam_client.get_role(RoleName=AUTHORIZER_IAM_ROLE_NAME)
+        raise RuntimeError(f"Expected {AUTHORIZER_IAM_ROLE_NAME} role to not exist."
+                           f" Did previous test cleanup successfully?")
+    except iam_client.exceptions.NoSuchEntityException:
+        pass
+
+    resp = iam_client.create_role(
+        RoleName=AUTHORIZER_IAM_ROLE_NAME,
```

Comment:
I'm a little concerned about reusing the same resource names. In the SageMaker tests create a new role, with a randomised suffix, every time we bootstrap. This ensures we don't accidentally override another run (if running in parallel), or that tests don't fail when creating a resource that already exists.

Reply:
Very good suggestion, I will update it.